contracts/BatchExchangeViewer.sol

```diff
@@ -112,28 +112,29 @@ contract BatchExchangeViewer {
         uint16 maxPageSize
     ) public view returns (bytes memory elements, bool hasNextPage, address nextPageUser, uint16 nextPageUserOffset) {
         elements = new bytes(maxPageSize * AUCTION_ELEMENT_WIDTH);
-        uint256 elementCount = 0;
+        setLength(elements, 0);
```

Comment:
This confuses me. Why don't we just set the length once at the end? Is it to save a stack variable? We might also be able to replace `hasNextPage` with `unfilteredCount` (if we set it to pageSize initially outside of the loop).

Reply:
Yes, this is to save a stack variable. We basically use `elements.length / AUCTION_ELEMENT_WIDTH` to encode what was `elementCount`.

That sounds like a good idea! I'll give it a shot, but I'm not sure if it will work. I had tried removing `nextPageUser` and `nextPageUserOffest` and just reuse the `previousPageUser` and `previousPageUserOffset` but that didn't work :(

Reply:
It doesn't seem to work :cry:, removing `hasNextPage` and just using `unfilteredCount` does not seem to give me a local variable back.

Reply:
I see. I think it is because `hasNextPage` is part of the return data and thus doesn't count as a variable. A next level workaround would be to collect local variables in a single memory helper struct and access it through this. E.g.

```solidity
struct Helper {
  bytes unfiltered;
  bytes element;
  uint256 gasLeftBeforePage;
  uint256 unfilteredCount;
}
```

Maybe a bit overkill for now but something to keep in mind if we run into the same issue in a future change.